src/service/layers-impl.js

```diff
@@ -658,12 +755,66 @@ export class LayoutElement {
    * @return {!PositionDef}
    */
   getOffsetFromParent() {
-    if (this.needsRemeasure_) {
-      this.remeasure();
-    }
+    this.remeasure();
     return this.position_;
   }
 
+  /**
+   * Whether the layout is a descendant of the most recently scrolled layer.
+   *
+   * This MUST NOT be called unless inside an #ancestry iterator function.
+   * "Activeness" is only determined for the lifespan of the #ancestry call.
+   *
+   * @return {boolean}
+   */
+  isActive() {
+    return dev().assertBoolean(this.isActive_);
+  }
+
+  /**
+   * Gets the minimal horizontal distance of this element from it's parent's
+   * "viewport".
+   */
+  getHorizontalDistanceFromParent() {
+    const parent = this.getParentLayer();
+    if (!parent) {
+      return 0;
+    }
+
+    const {left} = this.getOffsetFromParent();
+    const {width} = this.getSize();
+    const scrollLeft = parent.getScrollLeft();
+    const parentWidth = parent.getSize().width;
+    const distance = Math.max(
+        0,
+        scrollLeft - (left + width),
+        left - (scrollLeft + parentWidth)
+    );
+    return distance / parentWidth;
+  }
+
+  /**
+   * Gets the minimal vertical distance of this element from it's parent's
+   * "viewport".
+   */
+  getVerticalDistanceFromParent() {
```

Comment:
Why not combine this with above to avoid dupe calls to `remeasure` in `calcLayoutScore_`?